modules/contracts/src.ts/services/ethService.ts

```diff
@@ -280,137 +318,172 @@ export class EthereumChainService extends EthereumChainReader implements IVector
-                channelAddress,
-                reason,
-              });
-            }
-            // Break out of the loop here, as the tx has been completed.
-            break;
-          } catch (e) {
-            // TODO: Maybe it would be more robust to have waitForConfirmation return undefined or something
-            // specific in the event of timeout, as opposed to using error comparison?
-
-            // Check if the error was a confirmation timeout.
-            if (e.message === ChainError.retryableTxErrors.ConfirmationTimeout) {
-              // Scale up gas by percentage as specified by GAS_BUMP_PERCENT.
-              this.log.info(
-                { channelAddress, reason },
-                "Tx timed out waiting for confirmation. Bumping gas price and reattempting.",
-              );
-              gasPrice = gasPrice.add(gasPrice.mul(GAS_BUMP_PERCENT));
-            } else {
-              // If we get any other error here, we classify this event as a tx failure and break out of the loop.
-              this.log.error({ method: "sendTxAndParseResponse", error: jsonifyError(e) }, "Transaction reverted.");
-              await this.store.saveTransactionFailure(channelAddress, response.hash, e.message);
-              this.evts[ChainServiceEvents.TRANSACTION_FAILED].post({
-                error: e,
-                channelAddress,
-                reason,
-              });
-              break;
-            }
-          }
+        // If response returns undefined, we assume the tx was not sent. This will happen if some logic was
+        // passed into txFn to bail out at the time of sending
+        if (!response) {
+          this.log.warn({ method, methodId, channelAddress, reason }, "Did not attempt tx");
+          return undefined;
         }
-        return response;
-      });
 
-      if (!response) {
-        return Result.ok(response);
-      }
+        // save to store
+        await this.store.saveTransactionResponse(channelAddress, reason, response);
+        this.evts[ChainServiceEvents.TRANSACTION_SUBMITTED].post({
+          response: Object.fromEntries(
+            Object.entries(response).map(([key, value]) => {
+              return [key, BigNumber.isBigNumber(value) ? value.toString() : value];
+            }),
+          ) as StringifiedTransactionResponse,
+          channelAddress,
+          reason,
+        });
 
-      // add completed function
-      return Result.ok({
-        ...response,
-        completed: async (confirmations?: number) => {
-          try {
-            const receipt = await response.wait(confirmations);
-            if (receipt.status === 0) {
-              return Result.fail(new ChainError(ChainError.reasons.TxReverted, { receipt }));
+        // completed callback should always return a receipt and abstract the waiting and bumping of gas prices
+
+        // use a promise here so that the whole confirmation is not held up in the queue
+        let tryNumber = 0;
```

Comment:
This iteration variable (?) is never changed

Reply:
Damn it you're right.